### PR TITLE
Migrate to finalize methods

### PR DESCRIPTION
**Pull request by dwoz**

### CODE REVIEW

File: mayflower/build/windows.py [hunk 1]
```diff
@@ -3,6 +3,11 @@
 import urllib.request
 from .common import *
 
+if sys.platform == "win32":
```
Comment (twangboy) [inline mayflower/build/windows.py:6]: Don't you need to import sys first?

**Review by twangboy — CHANGES_REQUESTED** [review]

**Comment by twangboy** [comment]
On Windows (and maybe others) we could probably make the zip file smaller by removing all the `__pycache__` directories.

**Comment by dwoz** [comment]
> On Windows (and maybe others) we could probably make the zip file smaller by removing all the `__pycache__` directories.

No need to worry about this at this time.

File: mayflower/build/common.py [hunk 2]
```diff
@@ -584,7 +571,115 @@ def install_sysdata(mod, destfile, buildroot, toolchain):
...
+        mod = __import__(str(modname))
+    finally:
+        os.chdir(cwd)
+        sys.path = path
+    dest = pymodules / "site-packages" / "mayflower-sysconfigdata.py"
+    install_sysdata(mod, dest, dirs.prefix, dirs.toolchain)
+
+    # Lay down site customize
+    bindir = pathlib.Path(dirs.prefix) / "bin"
+    sitecustomize = bindir.parent / "lib" / "python3.10" / "site-packages" / "sitecustomize.py"
```
[the remark this answers is not in the document]
Comment (dwoz) [reply mayflower/build/common.py:598]: This  code doesn't run on windows.

File: mayflower/build/common.py [hunk 3]
```diff
@@ -584,7 +571,115 @@ def install_sysdata(mod, destfile, buildroot, toolchain):
...
+    runtime = MODULE_DIR / "common.py"
+    dest = mayflowerdir / "common.py"
+    with io.open(runtime, "r") as rfp:
+        with io.open(dest, "w") as wfp:
+            wfp.write(rfp.read())
+    init = mayflowerdir / "__init__.py"
+    init.touch()
+
+    # Install pip
+    python = dirs.prefix / "bin" / "python3"
```
[the remark this answers is not in the document]
Comment (dwoz) [reply mayflower/build/common.py:619]: This code doesn't run on windows.

File: mayflower/build/common.py [hunk 4]
```diff
@@ -584,7 +571,115 @@ def install_sysdata(mod, destfile, buildroot, toolchain):
...
+    if env["MAYFLOWER_ARCH"] != "x86_64":
+        env["MAYFLOWER_CROSS"] = dirs.prefix
+    runcmd([env["MAYFLOWER_NATIVE_PY"], "-m", "ensurepip"], env=env, stderr=logfp, stdout=logfp)
+
+
+    # XXX Is fixing shebangs still needed?
+    # Fix the shebangs in python's scripts.
+    bindir = pathlib.Path(dirs.prefix) / "bin"
+    pyex = bindir / "python3.10"
```
[the remark this answers is not in the document]
Comment (dwoz) [reply mayflower/build/common.py:628]: This code doesn't run on windows.

File: mayflower/build/common.py [hunk 5]
```diff
@@ -584,7 +571,115 @@ def install_sysdata(mod, destfile, buildroot, toolchain):
...
+
+    def runpip(pkg):
+        pip = bindir / "pip3"
+        target = None
+        #XXX This needs to be more robust
+        if sys.platform == "linux":
+            if env["MAYFLOWER_ARCH"] != "x86_64":
+                target = dirs.prefix / "lib" / "python3.10" / "site-packages"
+        cmd =  [
+            env["MAYFLOWER_NATIVE_PY"],
```
[the remark this answers is not in the document]
Comment (dwoz) [reply mayflower/build/common.py:662]: This code doesn't run on windows.

**Pull request merged**